modules/test/dns/python/src/dns_module.py

```diff
@@ -213,46 +212,46 @@ def _dns_network_from_dhcp(self):
 
     if dns_packets_local or dns_packets_not_local:
       if dns_packets_not_local:
-        result = False, 'DNS traffic detected to non-DHCP provided server'
+        # Feature Not Detected
+        result = ('Feature Not Detected',
+                  'DNS traffic detected to non-DHCP provided server')
       else:

```

Comment:
I believe this is still a failure scenario, DNS traffic is detected but not using the DHCP provided DNS servers.  No DNS traffic is captured as a feature not present.

Reply:
I would suggest the opposite. If the device is sending no DNS at all, then I would argue that this could be either Feature Not Detected (because it has not detected the feature of using local DNS servers) or Non-Compliant (because no DNS is considered Non Compliant. However, if the device is sending DNS, but to other DNS servers, this must be a Feature Not Detected as DNS is compliant and the feature of using local DNS servers is not detected (configured or present) on the device.

Reply:
I think you  might be conflating the two DNS tests into one when making those determinations.  Hostname resolution is what should only care that DNS was used to determine compliant/non-compliant or feature not detected.  DNS local is very clearly defined by the expected behavior:

        "test_description": "Verify the device allows for a DNS server to be entered automatically", 
        "expected_behavior": "The device sends DNS requests to the DNS server provided by the DHCP server",

If a device is using DNS to a remote DNS server, that's a clear failure, not a feature not detected.  This can only be feature not detected if no DNS is detected.  Otherwise, if DNS is present and not to the local, it fails because it is using DNS just not as defined.

Reply:
This does not line up with the definition of a recommended, required if applicable or optional test. These tests should never return Non-Compliant because that indicates that the device under test is a non-compliant device, which it is not.

Reply:
A test result should be independent of the requirement of the test.  The overall compliance of the device will take into account the requirement of the individual tests and the result to calculate it's final score.  In this case, this is a roadmap/recommended test which would still pass on compliance if the test fails while indicating it is not complaint to this specific test.  I don't see that as an issue.